src/state_space/mod.rs

```diff
@@ -213,13 +207,13 @@ impl StateSpace {
         self.state.get_mut(address)
     }
 
-    pub fn sync(&mut self, logs: &[Log]) -> Vec<Address> {
+    pub fn sync(&mut self, logs: &[Log]) -> Result<Vec<Address>, StateSpaceError> {
         let latest = self.latest_block.load(Ordering::Relaxed);
         let mut block_number = logs
             .first()
-            .expect("TODO: handle error")
+            .ok_or(StateSpaceError::MissingLogs)?
```

Comment:
We should use 

```rust

let block_number  = if let Some(log){
    // --snip--
} else{
    return Ok(vec![]);
}
```

Then we can remove `StateSpaceError::MissingLogs` from `StateSpaceError`